File: src/ZohoDocsSDK/ZohoDocsSDK/JSON/JSON_SharesMetadata.vb
Public Class JSON_SharesMetadata
    Public Property response As JSON_SharedDetails_Response

    Public ReadOnly Property MailsList As List(Of String)
        Get
            Dim co_Owner = response.result.coOwner.sharedUsers.Split(",").ToList()
            Dim Read_Only = response.result.ReadOnly.sharedUsers.Split(",").ToList()
            Dim Read_Write = response.result.ReadWrite.sharedUsers.Split(",").ToList()
            Dim all = co_Owner.Concat(Read_Only).Concat(Read_Write).Distinct().ToList() 'merege all lists
            Return all.Where(Function(s) Not String.IsNullOrEmpty(s)).ToList() 'remove empty
        End Get
    End Property

    Public ReadOnly Property SharedFileUrl As String
        Get
            Return response.result.permaLink
        End Get
    End Property

    Enum SharedORPublic
        [Shared]
        [Public]
    End Enum
    Public ReadOnly Property Shared_OR_Public As SharedORPublic
        Get
            If response.result.visibility = "private" Then
                Return SharedORPublic.Shared
            ElseIf response.result.visibility = "linkshare" Then
                Return SharedORPublic.Public
            End If
        End Get
    End Property
End Class
Public Class JSON_SharedDetails_Response
    Public Property result As JSON_SharedDetails_Result
    Public Property uri As String
    Public Property message As String
End Class
Public Class JSON_SharedDetails_Result
    Public Property [ReadOnly] As JSON_SharedDetails_Readonly
    Public Property permaLink As String
    Public Property coOwner As JSON_SharedDetails_Coowner
    Public Property visibility As String
    Public Property ReadComment As JSON_SharedDetails_Readcomment
    Public Property permission As String
    Public Property ReadWrite As JSON_SharedDetails_Readwrite
End Class
Public Class JSON_SharedDetails_Readonly
    Public Property sharedUsers As String
    Public Property isDirectlyShared As String
    Public Property sharedUserZuids As String
    Public Property sharedGroups As String
    Public Property sharedEmails As String
    Public Property isDirectlyEmailShared As String
    Public Property sharedGroupNames As String
    Public Property isDirectlyGroupShared As String
End Class
Public Class JSON_SharedDetails_Coowner
    Public Property sharedUsers As String
    Public Property isDirectlyShared As String
    Public Property sharedUserZuids As String
    Public Property sharedEmails As String
    Public Property isDirectlyEmailShared As String
End Class
Public Class JSON_SharedDetails_Readcomment
    Public Property sharedUsers As String
    Public Property sharedUserZuids As String
    Public Property sharedGroups As String
    Public Property sharedEmails As String
    Public Property sharedGroupNames As String
    Public Property isDirectlyGroupShared As String
End Class
Public Class JSON_SharedDetails_Readwrite
    Public Property sharedUsers As String
    Public Property isDirectlyShared As String
    Public Property sharedUserZuids As String
    Public Property sharedGroups As String
    Public Property sharedEmails As String
    Public Property isDirectlyEmailShared As String
    Public Property sharedGroupNames As String
    Public Property isDirectlyGroupShared As String
End Class

File: ZohoDocsSDK/ZohoDocsSDK/JSON/JSON_SharesMetadata.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ZohoDocsSDK.JSON
{
    public class JSON_SharesMetadata
    {
        public JSON_SharedDetails_Response response { get; set; }

        public List<string> MailsList
        {
            get
            {
                var co_Owner = response.result.coOwner.sharedUsers.Split(',').ToList();
                var Read_Only = response.result.ReadOnly.sharedUsers.Split(',').ToList();
                var Read_Write = response.result.ReadWrite.sharedUsers.Split(',').ToList();
                var all = co_Owner.Concat(Read_Only).Concat(Read_Write).Distinct().ToList(); // merege all lists
                return all.Where(s => !string.IsNullOrEmpty(s)).ToList(); // remove empty
            }
        }

        public string SharedFileUrl
        {
            get
            {
                return response.result.permaLink;
            }
        }

      public enum SharedORPublic
        {
            Shared,
            Public,
            none
        }
        public SharedORPublic Shared_OR_Public
        {
            get
            {
                switch (response.result.visibility)
                {
                    case "private":
                        return SharedORPublic.Shared;
                    case "linkshare":
                        return SharedORPublic.Public;
                    default : return  SharedORPublic.none;
                }
            }
        }
    }
    public class JSON_SharedDetails_Response
    {
        public JSON_SharedDetails_Result result { get; set; }
        public string uri { get; set; }
        public string message { get; set; }
    }
    public class JSON_SharedDetails_Result
    {
        public JSON_SharedDetails_Readonly ReadOnly { get; set; }
        public string permaLink { get; set; }
        public JSON_SharedDetails_Coowner coOwner { get; set; }
        public string visibility { get; set; }
        public JSON_SharedDetails_Readcomment ReadComment { get; set; }
        public string permission { get; set; }
        public JSON_SharedDetails_Readwrite ReadWrite { get; set; }
    }
    public class JSON_SharedDetails_Readonly
    {
        public string sharedUsers { get; set; }
        public string isDirectlyShared { get; set; }
        public string sharedUserZuids { get; set; }
        public string sharedGroups { get; set; }
        public string sharedEmails { get; set; }
        public string isDirectlyEmailShared { get; set; }
        public string sharedGroupNames { get; set; }
        public string isDirectlyGroupShared { get; set; }
    }
    public class JSON_SharedDetails_Coowner
    {
        public string sharedUsers { get; set; }
        public string isDirectlyShared { get; set; }
        public string sharedUserZuids { get; set; }
        public string sharedEmails { get; set; }
        public string isDirectlyEmailShared { get; set; }
    }
    public class JSON_SharedDetails_Readcomment
    {
        public string sharedUsers { get; set; }
        public string sharedUserZuids { get; set; }
        public string sharedGroups { get; set; }
        public string sharedEmails { get; set; }
        public string sharedGroupNames { get; set; }
        public string isDirectlyGroupShared { get; set; }
    }
    public class JSON_SharedDetails_Readwrite
    {
        public string sharedUsers { get; set; }
        public string isDirectlyShared { get; set; }
        public string sharedUserZuids { get; set; }
        public string sharedGroups { get; set; }
        public string sharedEmails { get; set; }
        public string isDirectlyEmailShared { get; set; }
        public string sharedGroupNames { get; set; }
        public string isDirectlyGroupShared { get; set; }
    }
}
